### ISSUE TITLE
Is the Dictionary type necessary?

### ISSUE THREAD

**Issue by s1na** [ethereumjs/rlp#62]
The `Input` definition includes a `Dictionary` type, even though `Dictionary` doesn't seem to be used in the code and `toBuffer` doesn't explicitly handle it. The RLP reference hasn't mentioned anything about objects either. I was wondering if it can be removed, or there's a reason why it's there?

**Comment by alcuadrado** [comment]
I think `Dictionary` is not only not needed, but incorrect. Calling `encode({a: 1123})` throws `invalid type`.

I believe it was added in [e81acfd9571](https://github.com/ethereumjs/rlp/commit/e81acfd9571e2e3a32e62e31a6fc47195ad12911) when trying to solve #35. TS support for recursive types is non-obvious, and while googling about it I got to [this comment by Anders Hejlsberg
](https://github.com/Microsoft/TypeScript/issues/3496#issuecomment-128553540) which looks almost the same as the original `types.ts`. This was probably copy-pasted and accidentally left here.
